src/modules/navigator/takeoff.cpp

```diff
@@ -96,62 +96,48 @@ Takeoff::on_active()
 void
 Takeoff::set_takeoff_position()
 {
-	struct position_setpoint_triplet_s *rep = _navigator->get_takeoff_triplet();
-
-	float abs_altitude = 0.0f;
-
-	float min_abs_altitude;
-
-	// TODO: review this, comments are talking about home pos, the validity is checked but the
-	// current altitude is used instead. Also, the "else" case does not consider the current altitude at all.
-	if (_navigator->home_alt_valid()) { //only use home position if it is valid
-		min_abs_altitude = _navigator->get_global_position()->alt + _navigator->get_takeoff_min_alt();
+	if (!_navigator->home_alt_valid()) {
+		mavlink_log_info(_navigator->get_mavlink_log_pub(),
+				 "Home altitude not valid, abort takeoff\t");
```

Comment:
This will most likely break auto takeoff with optical flow only.
That's maybe also the reason why home position use was removed historically (I didn't check).
Could we do one case with home position that does the right thing like you probably suggest and tested and one case without where it just does a local takeoff?

Reply:
Let's just verify that optical flow only takeoff still works. There's a simualtion target for this. Not sure if it's ran in CI.
https://docs.px4.io/main/en/simulation/#simulation
```
# Start Gazebo Classic with iris and optical flow
make px4_sitl gazebo-classic-classic_iris_opt_flow
```

Reply:
Right, I put back the ability to takeoff without Home set: adb01bf112ccad2ba976d2b736aac59c0a716dec

Reply:
@MaEtUgR It seems those make commands were wrong, see https://github.com/PX4/PX4-user_guide/pull/2709

Reply:
omg, I didn't even realize. Thanks for testing and fixing 🙏